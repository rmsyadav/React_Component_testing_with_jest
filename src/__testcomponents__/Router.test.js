import {render,screen} from "@testing-library/react";
import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import MyRoutes from "../Components/RoutesComponents/Router";

describe("Routing testing of components",()=>{
    
    it("snapshot testing of Router component",()=>{
     
        const Router = renderer.create(<BrowserRouter><MyRoutes></MyRoutes></BrowserRouter>).toJSON();
        expect(Router).toMatchSnapshot();

    })

    it("Particular route testing",()=>{
        render(
            <MemoryRouter initialEntries={['/']}>
               <MyRoutes></MyRoutes>
            </MemoryRouter>
          );
      const todo= screen.getByText("Todo example");
      expect(todo).toBeInTheDocument();
    });

    it("Particular route testing",()=>{
        render(
            <MemoryRouter initialEntries={['/list']}>
               <MyRoutes></MyRoutes>
            </MemoryRouter>
          );
      const list= screen.getByText("Numbered List Group");
      expect(list).toBeInTheDocument();
    });
})
