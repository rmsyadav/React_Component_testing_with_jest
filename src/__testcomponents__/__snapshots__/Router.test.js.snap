// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routing testing of components snapshot testing of Router component 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-warning bg-warning"
  >
    <div
      className="container-fluid"
    >
      <a
        className="text-sucess"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ms-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active active"
              data-testid="home"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              data-testid="list"
              href="/list"
              onClick={[Function]}
            >
              List
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              data-testid="signin"
              href="/signin"
              onClick={[Function]}
            >
              SignIn
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    class="container"
  >
    <div
      class="card text-center"
    >
      <div
        class="card-header"
      >
        Todo App
      </div>
      <div
        class="card-body"
      >
        <div
          class="input-group card-title mb-3"
        >
          <input
            class="form-control"
            onChange={[Function]}
            placeholder="Add your new todo"
            type="text"
          />
          <button
            class="btn btn-outline-secondary"
            id="button-addon2"
            onClick={[Function]}
            type="button"
          >
            <i
              class="bi bi-plus-circle-fill"
              style={
                Object {
                  "color": "blue",
                }
              }
            />
          </button>
        </div>
        <div
          class="card-text"
        >
          <ul
            class="list-group list-group-numbered"
          />
        </div>
      </div>
      <div
        class="card-footer text-muted"
      >
        Sun Dec 17 2023 03:24:00 GMT+0530 (India Standard Time)
      </div>
    </div>
  </div>,
]
`;
